### ISSUE TITLE
Getting redirected to login page on session expire

### ISSUE THREAD

**Issue by akkivaghasiya5** [ohmybrew/laravel-shopify#60]
My Shopify app is getting redirected to the login screen if the session is expired. Redirect to login screen is used for websites. But here, it's an app installed in a store. So if the session is getting expired, either it should be auto logged in somehow or should be displayed some message of session expire a redirected to app list. I have tried to override AuthShop middleware. But not getting the idea about what to do in the handle.

Here is the screenshot when the user gets redirected to the login screen.

![download](https://user-images.githubusercontent.com/19259101/43628890-2b41ee64-9719-11e8-9083-23e3c69c75e7.png)

**Comment by ohmybrew** [comment]
I'm not able to reproduce this. Are you using ngork or something?

**Comment by tobiasdalhof** [comment]
I'm working on a Shopify app which uses vuejs (SPA) and in order to keep the session fresh, I implemented a simple solution with JS.

```js
// Keep the session fresh
const ping = setInterval(() => {
  // userApi.ping() is basically axios.get('api/user/ping')
  // api/user/ping route uses the 'auth.shop' middleware
  userApi.ping().then(response => {
    if (
      typeof response.data.status === undefined ||
      response.data.status !== 200
    ) {
      clearInterval(ping)
      window.location.reload(false)
    }
  }).catch(err => {
    clearInterval(ping)
    window.location.reload(false)
  })
}, 60000*5)
```

```php
<?php
// ..

class UserController extends Controller
{
    // ...

    /**
     * Keep the session fresh.
     */
    public function ping()
    {
        return response()->json([
            'status' => 200,
        ], 200);
    }

    // ...
}
```

**Comment by ohmybrew** [comment]
I'll see why it's redirecting to login page.. it should go back to auth and
redo the flow..

**Comment by ptcampbell** [comment]
This happens to me too. Have not looked enough into the why, but I will try something similar to @tobiasdalhof's solution with React.

**Comment by ohmybrew** [comment]
I'm guessing the Laravel session is expiring. Then hitting the route, it
goes to auth but has no shop so it redirects to login... however there
should still be a shop in the parameters of the query string... I'll try to
reproduce.

**Comment by ohmybrew** [comment]
Okay all, I did some digging. This is not a package issue it seems but a Laravel thing...

Checking out `shopify_app` gem which I use for some projects, which this package was originally based on, it sets a cookie via Rails which contains the session information, this cookie *expires on browser close only*, it has a date of 1969 for expire.

Testing the above, I deleted the cookie (invalidating the session), `shopify_app` does the same thing, displays a login/install page.

---

Laravel, setting a session creates a cookie as well, *but expires 120 minutes as default*.

Change `config/session.php`....

```
# config/session.php
'expire_on_close' => true,
```

We get the same result, a cookie `laravel_session` with a date of 1969. I left my browser overnight, went back, clicked a link on the app, and things still worked. I think this may be the issue, can someone else confirm? If so, maybe I need to add this as a install step...

**Comment by ptcampbell** [comment]
Thanks Tyler. I’ll give it a shot.

**Comment by ohmybrew** [comment]
I believe this is the issue yes, I just clicked a link again on a browser tab I did not close which contained the app, it was fine. Another tab without the modification to expire_on_close, it kicked me out. I think I'll update the docs on this for install.

**Comment by ohmybrew** [comment]
After some fiddle with the code, it seems I can override Laravel setting safely during auth, so I will move forward with a code addition to assist in this.

**Comment by ohmybrew** [comment]
Should be solved now with v3.0.2, I had a view-source "watcher" extension save the redirects, and can confirm its now no longer encoded.
